bson/bsonrw/extjson_wrappers.go

```diff
@@ -95,12 +96,14 @@ func (ejv *extJSONValue) parseBinary() (b []byte, subType byte, err error) {
 				return nil, 0, fmt.Errorf("$binary subType value should be string, but instead is %s", val.t)
 			}
 
-			i, err := strconv.ParseInt(val.v.(string), 16, 64)
+			i, err := strconv.ParseUint(val.v.(string), 16, 64)
 			if err != nil {
 				return nil, 0, fmt.Errorf("invalid $binary subType string: %s", val.v.(string))
 			}
 
-			subType = byte(i)
+			b := []byte{0, 0, 0, 0, 0, 0, 0, 0}
+			binary.LittleEndian.PutUint64(b, i)
+			subType = b[0]
```

Comment:
We can use `ParseUint` to limit the range of the parsed subtype value by using `bitSize=8`. I also recommend adding the parse error to returned error.

E.g.
```go
i, err := strconv.ParseUint(val.v.(string), 16, 8)
if err != nil {
	return nil, 0, fmt.Errorf("invalid $binary subType string %q: %w", val.v.(string), err)
}
subType = byte(i)
```